### PR TITLE
CGI modules

### PR DESCRIPTION
**Pull request by edouard-sn**
# Description

This PR implements CGIs modules. (php, ruby and perl).

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [ ] I have tested this code
- [X] I have added sufficient documentation in the code

### CODE REVIEW

File: modules/include/cgi/ACGI.hpp [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+#define FGETS fgets
+#else
+#define POPEN popen
+#define PCLOSE pclose
+#define FGETS fgets
+#endif
+
+class ACGI : public IHandlerModule {
+protected:
+    static const inline std::string kHeaderBodySeparator = "\r\n\r\n";
```
Comment (martin-olivier) [inline modules/include/cgi/ACGI.hpp:34]: maybe use `std::string_view` instead
```suggestion
    static constexpr std::string_view kHeaderBodySeparator = "\r\n\r\n";
```

**Review by martin-olivier — APPROVED** [review]
Looks great !

File: modules/include/cgi/ACGI.hpp [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+    void Init(const config::Node &cfg)
+    {
+        const auto get_cfg_string = [&cfg, this](const std::string &key, std::string &variable) {
+            try {
+                auto &dict = cfg["modules"][GetName()].AsDict();
+
+                variable = dict.at(key)->AsString();
+            } catch (const std::exception &) {
+                Logger::Warning(std::string(GetName()) + ": Couldn't load " + key + ". A default value will be used (" +
+                                variable + ").");
```
Comment (rojasdiegopro) [inline modules/include/cgi/ACGI.hpp:47]: Le `ziapi::Logger` gère les arguments variadique et permet de print tout ce que tu peux print dans un stream.
Tu pourrais remplacer par `Logger::Warning(GetName(), ": Couldn't load");` etc...

**Review by rojasdiegopro — APPROVED** [review]
Je suis grave impressed. Super clean et tout. Je peux pas garantir avoir tout compris par contre :hihi:

**Pull request merged**